src/modules/RA/User/components/UserFields.js

```diff
@@ -1,35 +1,79 @@
 import React from 'react';
-
 import {
   TextInput,
   SimpleForm,
   BooleanInput,
   ReferenceArrayInput,
   SelectArrayInput,
+  useTranslate,
 } from 'react-admin';
 
+import FormGroup from '@material-ui/core/FormGroup';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Switch from '@material-ui/core/Switch';
+import Typography from '@material-ui/core/Typography';
+import { connectAuthProvider } from '@terralego/core/modules/Auth';
+
 import JSONInput from '../../../../components/react-admin/JSONInput';
+import FieldGroup from '../../../../components/react-admin/FieldGroup';
 import UserFieldsHelp from './UserFieldsHelp';
 
 import { RES_USERGROUP } from '../../ra-modules';
 
-const UserFields = ({ edit = false, ...props }) => (
-  <SimpleForm {...props}>
-    {edit && <TextInput disabled source="id" />}
-    {edit && <TextInput disabled source="uuid" />}
+const UserFields = ({ edit = false, user: { is_superuser: isSuperUser } = {}, ...props }) => {
+  const [advanced, setAdvanced] = React.useState(false);
+  const translate = useTranslate();
+
+  const onAdvancedChange = ({ target: { checked } }) => {
+    setAdvanced(checked);
+  };
+
+  return (
+    <SimpleForm {...props}>
+      {edit && <TextInput disabled source="id" />}
+      {edit && <TextInput disabled source="uuid" />}
+
+      <TextInput source="properties.first_name" label="user.form.firstname" />
+      <TextInput source="properties.last_name" label="user.form.lastname" />
+      <TextInput source="email" type="email" />
+      {!edit && <TextInput source="password" type="password" />}
+      { isSuperUser && <BooleanInput source="is_superuser" />}
```

Comment:
It looks good to me!
Maybe it's better to display the field but disabled in order to escort the user. Something like this ⬇️ 
```jsx
<BooleanInput 
  source="is_superuser" 
  disabled={!isSuperUser}
  helperText={!isSuperUser && 'You don't have enough permission to edit this field' }
/>
```
 

```

Reply:
Done!